### PR TITLE
RFC Tidy test pins

### PR DESCRIPTION
**Pull request by robtaylor**
This is a request for comments.

The nonlocality of led_port_pin_map bugged me, so hacked this approach up whilse bored, which adds moves the definitions of test output pins to the board defintions.

### CODE REVIEW

**Comment by cfelton** [comment]
@robtaylor I think you are correct, there is room for improvement here.  This originally started off as an example for a workshop and the `blinky` grew into a generic first pass test for the automated build test, room for improvement indeed.

I am not sure I like the `test_output_pins` added to the board definitions because it is specific to the test, and not meaningful in most other cases.  I think the `test_output_pins` definitions (which port/pins to be used for the test) should be moved to the test file instead of test specific information in the board definition.

I probably should rework the example/build and use something other than blinky.  The reason, some else commented that having the `blinky` in each board directory is more intuitive to find and the explicit example is useful (although redundant and a maintenance pain).  Currently, I point folks to the blinky examples in the example/build.

For the near-term I think capturing the `test_output_pins` in test_boards.py and removing board_build_example.py would clear up some of the confusion (using your proposed `add_test_port`).  Long term replace `blinky` with an example design that has more meat (as in FPGA resources) but limit it to 3-4 pins (clock reset (optional), serialin, serialout).

**Comment by robtaylor** [comment]
ack, I can see the reasoning there, will tidy up to that effect

**Comment by cfelton** [comment]
I am back tracking on the removal of the `board_build_example.py`, it is a good example (being in the example directory) how to use the board defs to extract information on the board.  What needs to be done is that `get_port` should have some accept a glob syntax and more options: options to return closest match).

**Comment by robtaylor** [comment]
Hopefully this is is better :)

File: examples/build/board_build_example.py [hunk 1]
```diff
@@ -10,19 +10,7 @@
...
-# to a set of pins for boards without an led port.
-led_port_pin_map = {
-    'xula':  dict(name='led', pins=(36, 37, 50, 52)),
-    'xula2': dict(name='led', pins=('R7', 'R15', 'R16', 'M15',)),
-    'pone': dict(name='led', pins=(18, 23, 26, 33)),
-    # the de0-cv led port is named "ledr", for the generic testing
-    # add the led port
-    'de0cv': dict(name='led', pins=('AA2', 'AA1', 'W2', 'Y3',))
-}
+from rhea.build.boards import led_port_pin_map
```
Comment (cfelton) [inline examples/build/board_build_example.py]: Didn't you move this to test_boards.py?  Shouldn't this fail?

File: examples/build/ex_xula.py [hunk 2]
```diff
@@ -3,13 +3,13 @@
 
 import rhea.build as build
 from rhea.build.boards import get_board
+from rhea.build.boardsimport led_port_pin_map
```
Comment (cfelton) [inline examples/build/ex_xula.py]: typo

File: examples/build/test_boards.py [hunk 3]
```diff
@@ -10,9 +10,17 @@
...
 
+# The following maps pins to use for test output for boards without an led port.
+led_port_pin_map = {
+    'xula': dict(name='led', port='chan', slc=[slice(0,2),slice(3,2)]),
+    'xula2': dict(name='led', port='chan', slc=slice(0,4)),
+    'pone': dict(name='led', port='wingA', slc=slice(0,4)),
+    'de0cv': dict(name='led', port='ledr', slc=slice(0,4))
+}
+
+
```
Comment (cfelton) [inline examples/build/test_boards.py]: This works nice, it is more explicit which default port name is being used.

File: examples/build/board_build_example.py (hunk 1 above)
Comment (robtaylor) [inline examples/build/board_build_example.py]: On 05/11/15 20:25, Christopher Felton wrote:
> In examples/build/board_build_example.py
> <https://github.com/cfelton/rhea/pull/4#discussion_r44056522>:
> 
>> @@ -10,19 +10,7 @@
>>  import rhea.build as build 
>>  from rhea.build.boards import get_board
>>  from rhea.build.boards import get_all_board_names
>> -#from blink import blinky
>> -
>> -# some boards don't have LEDs but they have IO that
>> -# can drive LEDs.  The following maps the led port 
>> -# to a set of pins for boards without an led port.
>> -led_port_pin_map = {
>> -    'xula':  dict(name='led', pins=(36, 37, 50, 52)),
>> -    'xula2': dict(name='led', pins=('R7', 'R15', 'R16', 'M15',)),
>> -    'pone': dict(name='led', pins=(18, 23, 26, 33)),
>> -    # the de0-cv led port is named "ledr", for the generic testing
>> -    # add the led port
>> -    'de0cv': dict(name='led', pins=('AA2', 'AA1', 'W2', 'Y3',))
>> -}
>> +from rhea.build.boards import led_port_pin_map
> 
> Didn't you move this to test_boards.py? Shouldn't this fail?
> 

This is the removement, not sure I understand the question!

Rob

File: examples/build/ex_xula.py (hunk 2 above)
Comment (robtaylor) [inline examples/build/ex_xula.py]: On 05/11/15 20:25, Christopher Felton wrote:
> In examples/build/ex_xula.py
> <https://github.com/cfelton/rhea/pull/4#discussion_r44056545>:
> 
>> @@ -3,13 +3,13 @@
>>  
>>  import rhea.build as build
>>  from rhea.build.boards import get_board
>> +from rhea.build.boardsimport led_port_pin_map
> 
> typo
> 

Good catch, next version on its way

Rob

File: examples/build/board_build_example.py (hunk 1 above)
Comment (cfelton) [inline examples/build/board_build_example.py]: The comment was on the import, you didn't add anything to `rhea.build.boards` the import fails, the import needs to change to `from test_boards import led_port_pin_map`.  Not sure if that was the goal of your changes?

**Pull request merged**